cpp/ql/src/semmle/code/cpp/dataflow/internal/TaintTrackingUtil.qll

```diff
@@ -30,6 +31,11 @@ predicate localTaintStep(DataFlow::Node src, DataFlow::Node sink) {
  */
 predicate defaultAdditionalTaintStep(DataFlow::Node src, DataFlow::Node sink) {
   localAdditionalTaintStep(src, sink)
+  or
+  exists(FunctionCall call, int i |
+    sink.(DataFlow::IteratorPartialDefinitionNode).getPartialDefinition().definesExpressions(_, call.getArgument(i)) and
+    src.(DataFlow::RefParameterFinalValueNode).getParameter() = call.getTarget().getParameter(i)
+  )
```

Comment:
We don't usually want to add interprocedural steps here. It means we lose the calling context and that we're not using the virtual-dispatch library.

I'm afraid this is only a quick drive-by review; remind me to follow up.

Reply:
Oh - in that case these steps may need to be in `DataFlow` rather than `TaintTracking`, which means reverting fa0e27b and separating `DefinitionByReferenceNode` and `RefOutNode`